internal/terraform/transform_config.go

```diff
@@ -170,6 +171,12 @@ func (t *ConfigTransformer) transformSingle(g *Graph, config *configs.Config, ge
 	// TODO: We could actually catch and process these kind of problems earlier,
 	//   this is something that could be done during the Validate process.
 	for _, i := range importTargets {
+		// The case in which an unmatched import block targets an expanded
+		// resource instance can error here. Others can error later.
+		if i.Addr.Resource.Key != nil {
```

Comment:
Thanks. Looks like https://github.com/hashicorp/terraform/blob/a801ba45cb454dea48c9ef4b1713ec1cea985594/internal/terraform/context_plan.go#L531 should be changed too then.

Reply:
`addrs.NoKey` literally _is_ the `nil` value of `addrs.InstanceKey`, so which to use is just a matter of style rather than function.

I've typically used variables like `addrs.NoKey` to help communicate that the `nil` value is valid for this type (for many interfaces it wouldn't be) and what that value represents. I don't feel super strongly about how we use it in comparisons like this, though perhaps `addrs.NoKey` here does make it _marginally_ more readable by naming what a `nil` `InstanceKey` actually represents. :man_shrugging: I wouldn't suggest we go and change all of the references just for the sake of changing them, but if we're "in the neighborhood" anyway then it's always nice to make things a little more consistent along the way.